appcues/src/main/java/com/appcues/ui/composables/AppcuesComposition.kt

```diff
@@ -114,47 +111,52 @@ private fun BoxScope.ComposeLastRenderingState(state: Rendering) {
         val metadataSettingTraits = remember(state.position) { mutableStateOf(steps[state.position].metadataSettingTraits) }
         val previousStepMetaData = remember { mutableStateOf(AppcuesStepMetadata()) }
         val stepMetadata = remember(metadataSettingTraits.value) {
-
-            val actual = hashMapOf<String, Any?>().apply { metadataSettingTraits.value.forEach { putAll(it.produceMetadata()) } }
-
-            mutableStateOf(AppcuesStepMetadata(previous = previousStepMetaData.value.actual, actual = actual)).also {
-                previousStepMetaData.value = it.value
+            try {
```

Comment:
maybe move this remember block to a function. called

rememberStepmetadata(metadataSettingTraits, onTraitException = { 
    viewModel.onTraitException(ex)
    return@MainSurface (something like that)
})

Reply:
if its not onTraitException the return value is not null

Reply:
I'm not sure I follow exactly yet. I can extract to a function, but the behavior right now does return a non-null value when it is not the exception case. The exception case returns a null value, which is then later used to prevent the rest of the composition from attempting to render with invalid metadata.